analyzer/queries/queries.go

```diff
@@ -437,35 +443,76 @@ const (
       runtime = $1 AND
       token_address = $2`
 
-	RuntimeEVMTokenBalanceAnalysisStale = `
-    SELECT
-      evm_token_balance_analysis.token_address,
-      evm_token_balance_analysis.account_address,
-      evm_tokens.token_type,
-      evm_token_balances.balance,
-      token_address_preimage.context_identifier,
-      token_address_preimage.context_version,
-      token_address_preimage.address_data,
-      account_address_preimage.context_identifier,
-      account_address_preimage.context_version,
-      account_address_preimage.address_data,
-      (SELECT MAX(height) FROM chain.processed_blocks WHERE analyzer = evm_token_balance_analysis.runtime::text AND processed_time IS NOT NULL) AS download_round
-    FROM chain.evm_token_balance_analysis
-    -- No LEFT JOIN; we need to know the token's type to query its balance.
-    -- We do not exclude tokens with type=0 (unsupported) so that we can move them off the DB index of stale tokens.
-    JOIN chain.evm_tokens USING (runtime, token_address)
-    LEFT JOIN chain.evm_token_balances USING (runtime, token_address, account_address)
-    LEFT JOIN chain.address_preimages AS token_address_preimage ON
-      token_address_preimage.address = evm_token_balance_analysis.token_address
-    LEFT JOIN chain.address_preimages AS account_address_preimage ON
-      account_address_preimage.address = evm_token_balance_analysis.account_address
-    WHERE
-      evm_token_balance_analysis.runtime = $1 AND
-      (
-        evm_token_balance_analysis.last_download_round IS NULL OR
-        evm_token_balance_analysis.last_mutate_round > evm_token_balance_analysis.last_download_round
+	RuntimeEVMTokenBalanceAnalysisStale = fmt.Sprintf(`
+    WITH
+    stale_evm_tokens AS (
+      SELECT
+        analysis.token_address,
+        analysis.account_address,
+        evm_tokens.token_type,
+        evm_token_balances.balance,
+        token_preimage.context_identifier,
+        token_preimage.context_version,
+        token_preimage.address_data,
+        account_preimage.context_identifier,
+        account_preimage.context_version,
+        account_preimage.address_data,
+        (SELECT MAX(height) FROM chain.processed_blocks WHERE analyzer = analysis.runtime::text AND processed_time IS NOT NULL) AS download_round
```

Comment:
Small nit: since this query now already uses CTE it might make it a bit nicer if the download round query is moved out into a separate CTE.

Reply:
ok, done